cumulus/pallets/xcmp-queue/src/migration.rs

```diff
@@ -113,6 +149,36 @@ pub fn migrate_to_v3<T: Config>() -> Weight {
 	T::DbWeight::get().reads_writes(overweight_messages, 1)
 }
 
+pub fn migrate_to_v4<T: Config>() -> Weight {
+	let translate = |pre: Vec<v3::OutboundChannelDetails>| -> Vec<super::OutboundChannelDetails> {
+		pre.into_iter()
+			.map(|details| super::OutboundChannelDetails {
+				recipient: details.recipient,
+				state: details.state,
+				signals_exist: details.signals_exist,
+				first_index: details.first_index as u64,
+				last_index: details.last_index as u64,
+			})
+			.collect()
+	};
+
+	if OutboundXcmpStatus::<T>::translate(|pre| pre.map(translate)).is_err() {
+		log::error!(
+			target: super::LOG_TARGET,
+			"unexpected error when performing translation of the OutboundXcmpStatus type during storage upgrade to v4"
+		);
+	}
+
+	let mut weight = T::DbWeight::get().reads_writes(1, 1);
+
+	for (paraid, idx, msg) in v3::OutboundXcmpMessages::<T>::drain() {
```

Comment:
I dont know if we can do this, since it runs on para-chains…  
Maybe we can do a lazy migration? The `u16` and`u64` are non-compact I assume, so they should be distinguishable?

Reply:
Are the hashes of a u16 the same as a u64? If not, they'll have completely different storage prefixes.

Reply:
Ah ok, I just verified this myself, the hashes for `u16` and `u64` are indeed different, so it's impossible to do a lazy migration, sadly...

Reply:
So I just verified this -- the hashes are indeed different. How exactly can I migrate then? The code needs to know how to fetch data from the old map in order to lazily migrate...

Reply:
`storage_alias`?

Reply:
I think the easiest way is to define a new map with `u64` key and then first lookup in the old map, and if it fails then fallback to the new one.  
So it would lazily migrate them on access.

Reply:
Okay, I've created a new double map called `OutboundXcmpMessageQueue`, which contains `u64` indices instead of `u16`. This should do the trick.